stew/result.nim

```diff
@@ -175,7 +170,7 @@ type
 func raiseResultError[T, E](self: Result[T, E]) {.noreturn.} =
   mixin toException
 
-  when E is ref Exception:
+  when E is ref Defect or E is ref CatchableError:
```

Comment:
this change looks a bit overzealous - I mean, if someone's wrapping legacy code, they'll have a bunch of exceptions around, and we're not really inheriting from it..